File: FootballExplorer/app/src/main/java/fr/a42consulting/footballexplorer/Presenter/PlayersListPresenter.java
package fr.a42consulting.footballexplorer.Presenter;

import org.json.JSONException;
import org.json.JSONObject;

import fr.a42consulting.footballexplorer.HttpGetRequest;
import fr.a42consulting.footballexplorer.Model.PlayersListModel;
import fr.a42consulting.footballexplorer.View.IPlayersListView;

public class PlayersListPresenter implements IPlayersListPresenter {
    private IPlayersListView playersListView;
    private PlayersListModel team;

    public PlayersListPresenter(IPlayersListView listView) {
        this.playersListView = listView;
    }

    @Override
    public void onLoadPlayers(String request) {
        HttpGetRequest getRequest = new HttpGetRequest() {
            @Override
            public void onResponseReceived(String result) {
                if (result == null) {
                    team = new PlayersListModel();
                    playersListView.onRequestResult(team.getPlayers());
                    return;
                }
                try {
                    JSONObject jsonResult = new JSONObject(result);
                    team = PlayersListModel.fromJson(jsonResult);
                    playersListView.onRequestResult(team.getPlayers());
                } catch (JSONException e) {
                    playersListView.onMessage("La récupération des données a échoué.");
                    e.printStackTrace();
                }
            }
        };

        getRequest.execute("https://www.thesportsdb.com/api/v1/json/1/searchplayers.php", "t", request);
    }
}

File: FootballExplorer_kotlin/app/src/main/java/fr/a42consulting/footballexplorer/presenter/PlayersListPresenter.kt
package fr.a42consulting.footballexplorer.presenter

import org.json.JSONException
import org.json.JSONObject

import fr.a42consulting.footballexplorer.model.PlayersListModel
import fr.a42consulting.footballexplorer.view.IPlayersListView
import org.jetbrains.anko.doAsync
import org.jetbrains.anko.uiThread
import java.net.URL
import java.net.URLEncoder

class PlayersListPresenter(private var playersListView: IPlayersListView?) : IPlayersListPresenter {
    private var team: PlayersListModel? = null

    override fun onLoadPlayers(request: String) {
        doAsync {
            val result = URL("https://www.thesportsdb.com/api/v1/json/1/searchplayers.php?t=" + URLEncoder.encode(request, "UTF-8")).readText()
            uiThread {
                try {
                    val jsonResult = JSONObject(result)
                    team = PlayersListModel.fromJson(jsonResult)
                    playersListView!!.onRequestResult(team!!.players)
                } catch (e: JSONException) {
                    playersListView!!.onMessage("La récupération des données a échoué.")
                    e.printStackTrace()
                }
            }
        }
    }

    override fun resetView(view: IPlayersListView?) {
        playersListView = view
    }
}
